File: server/src/main/scala/dsmoq/exceptions/InputValidationException.scala
package dsmoq.exceptions

import scala.collection.mutable

class InputValidationException(errors: scala.collection.Map[String, String]) extends RuntimeException{
  val validationErrors = errors

  def getErrorMessage() = {
    validationErrors.map(x =>
      InputValidationError(
        name = x._1,
        message = x._2
      ))
  }
}

case class InputValidationError(
  name: String,
  message: String
)

File: sdk/src/main/java/dsmoq/sdk/util/InputValidationException.java
package dsmoq.sdk.util;

import java.util.ArrayList;
import java.util.List;

public class InputValidationException extends RuntimeException {
    private List<InputValidationError> errors;
    public InputValidationException(List<InputValidationError> errors) {
        this.errors = new ArrayList<>(errors);
    }

    public List<InputValidationError> getErrorMessages() {
        return new ArrayList<>(this.errors);
    }
}
